src/SeleniumLibrary/keywords/cookie.py

```diff
@@ -39,18 +39,29 @@ def delete_cookie(self, name):
         self.driver.delete_cookie(name)
 
     @keyword
-    def get_cookies(self):
+    def get_cookies(self, as_dict=False):
         """Returns all cookies of the current page.
 
         The cookie information is returned as a single string in format
-        ``name1=value1; name2=value2; name3=value3``. It can be used,
-        for example, for logging purposes or in headers when sending
-        HTTP requests.
+        ``name1=value1; name2=value2; name3=value3`` if optional
+        ``as_dict`` is not provided or not explictly set to True.
+        Keyword can be used, for example, for logging purposes or in
+        headers when sending HTTP requests. In later case, setting
+        ``as_dict`` to True is helpful as then the result can be passed
+        to requests library's Create Session keyword's optional cookies
+        parameter.
+
         """
-        pairs = []
-        for cookie in self.driver.get_cookies():
-            pairs.append(cookie['name'] + "=" + cookie['value'])
-        return '; '.join(pairs)
+        if not as_dict:
+            pairs = []
+            for cookie in self.driver.get_cookies():
+                pairs.append(cookie['name'] + "=" + cookie['value'])
+            return '; '.join(pairs)
+        else:
+            pairs = {}
+            for cookie in self.driver.get_cookies():
+                pairs[cookie['name']] = cookie['value']
+            return pairs
```

Comment:
Instead of returning dictionary, it might be better to return RF dotdict https://github.com/robotframework/robotframework/blob/2.9.2/src/robot/utils/dotdict.py